lib/main.js

```diff
@@ -149,61 +165,92 @@ if either your review configuration or your project's dependencies change.`);
   exit(0);
 }
 
-module.exports = () => {
+/**
+ * @returns {Promise<void | never>}
+ */
+async function main() {
   if (options.version) {
     console.log(Anonymize.version(options));
     return;
   }
 
-  if (options.subcommand === 'init') {
-    if (options.help) {
-      return Help.init();
+  switch (options.subcommand) {
+    case 'init': {
+      if (options.help) {
+        return Help.init();
+      }
+
+      try {
+        return await Init.promptAndCreate(options);
+      } catch (error) {
+        errorHandler(error);
+      }
     }
 
-    return Init.promptAndCreate(options).catch(errorHandler);
-  }
+    case 'new-rule': {
+      if (options.help) {
+        return Help.newRule();
+      }
 
-  if (options.subcommand === 'new-rule') {
-    if (options.help) {
-      return Help.newRule();
+      try {
+        return await NewRule.create(options);
+      } catch (error) {
+        errorHandler(error);
+      }
     }
 
-    return NewRule.create(options).catch(errorHandler);
-  }
+    case 'new-package': {
+      if (options.help) {
+        return Help.newPackage();
+      }
 
-  if (options.subcommand === 'new-package') {
-    if (options.help) {
-      return Help.newPackage();
+      try {
+        return await NewPackage.create(options);
+      } catch (error) {
+        errorHandler(error);
+      }
     }
 
-    return NewPackage.create(options).catch(errorHandler);
-  }
+    case 'suppress': {
+      if (options.help) {
+        return Help.suppress();
+      }
 
-  if (options.subcommand === 'suppress') {
-    if (options.help) {
-      return Help.suppress();
-    }
+      if (options.suppressCheckAfterTests) {
+        return SuppressedErrors.checkForUncommittedSuppressions(options);
+      }
 
-    if (options.suppressCheckAfterTests) {
-      return SuppressedErrors.checkForUncommittedSuppressions(options);
+      break;
     }
-  }
 
-  if (options.subcommand === 'prepare-offline') {
-    if (options.help) {
-      return Help.prepareOffline();
+    case 'prepare-offline': {
+      if (options.help) {
+        return Help.prepareOffline();
+      }
+
+      return await prepareOffline();
     }
 
-    return prepareOffline();
-  }
+    default: {
```

Comment:
Should this not be `undefined:`, so that we get exhaustiveness errors from TS when we add new subcommands?

Reply:
Hmmm... can you do that? I'll check in a few minutes.